src/NNlib.jl

```diff
@@ -5,10 +5,14 @@ using Requires
 include("dim_helpers.jl")
 
 # NNPACK support
-if Sys.islinux() || Sys.isapple()
-    include("nnpack/NNPACK.jl")
+if get(ENV, "NNLIB_USE_NNPACK", "false") == "true"
+  if Sys.islinux() || Sys.isapple()
```

Comment:
This is then something that needs to be set during precompile time. And if you want to change it, how do you cause the package to be recompiled for it to take affect?

Reply:
Good catch, maybe I should move this to an `__init__` function.

Reply:
That has some other negatives in the sense that the NNPack code will no longer be preconpilable. One way would probably be to not use dispatch to select the backend code but instead have something like

```
function conv(...)
    if get(ENV(...))
         _NNPack_conv(...)
    else
        generic_conv(...)
    end
end
```

But that would require some restructuring of code that might be a bit annoying. 

The easiest is probably to move to use a build step. The build file looks at the ENV variable and creates a file that includes the NNPack depending on the env variable. If one wants to change this option, one sets the nrw env var and the runs `build` on the package.

Reply:
You mean doing this in the `build.jl` file and write to `deps.jl`?

Reply:
Could you please take a look? I've made it so.